rosbag2_transport/include/rosbag2_transport/player.hpp

```diff
@@ -200,6 +203,10 @@ class Player : public rclcpp::Node
   void delete_on_play_message_callback(const callback_handle_t & handle);
 
 protected:
+  void init(
```

Comment:
Is `init` part of the component node API? Is there instead a design that all constructors bubble into one final common constructor signature? I generally prefer to avoid separate initialization (even if protected) - if possible, the constructor should initialize.